chore: lint on pre-commit hook

piuccio opened the pull request:
I've also removed mocha tests because they're simply testing things that eslint verifies already.

On teamcity, milo test browsers are not running (missing `test-browser` in `package.json`) so there's a failing test, I've disabled it.
As it turns out the behaviour is actually broken, I came to `milo-ui` to fix a problem with multiple dialogs at the same time, but I wanted to clean things first.

Richard-Walton commented:
@piuccio  - We need to talk about the multiple dialog problem.  I need that behaviour.

Commit:
chore: lint on pre-commit hook

I've also removed mocha tests because they're simply testing things that eslint verifies already
